src/client/lib/controllers/usersServices.dart

```diff
@@ -0,0 +1,130 @@
+import 'dart:convert';
+import 'dart:ffi';
+import 'package:http/http.dart' as http;
+import 'package:shared_preferences/shared_preferences.dart';
+
+class usersServices{
+
+  Future<bool> loginUser(String username, String password) async {
+
+    final url = Uri.parse('http://10.0.2.2:8080/users/$username/login');
+
+    final body = jsonEncode({
+      'username': username,
+      'password': password,
+    });
+
+    // Send POST request to /login
+    final response = await http.post(
+      url,
+      headers: {'Content-Type': 'application/JSON'},
+      body: body,
+    );
+
+    print(response.statusCode);
+    print(response.body);
+
+    if (response.statusCode == 200) {
+      final data = jsonDecode(response.body);
+      final sessionId = data['session_id'];
+
+      // Store the session ID in shared preferences
+      final prefs = await SharedPreferences.getInstance();
+      await prefs.setString('sessionId', sessionId);
+
+      return true;
+    } else {
+      return false;
+    }
+  }
+
+  Future<bool> logoutUser(String username) async {
```

Comment:
Why do we pass the username instead of storing it in shared preferences? why do we even need to pass the username in the first place if the session ID is uniquely tied to the user? This is a flaw in the API design. Passing just a session ID is sufficient to log out.

Reply:
My fault lapse of judgment while coding it is supposed to be session-id.